packages/fcl/src/wallet-provider-spec/draft-v3.md

```diff
@@ -0,0 +1,494 @@
+## Status
+
+- **Last Updated:** August 3rd 2021
+- **Stable:** Yes
+- **Risk of Breaking Change:** Medium
+- **Compatibility:** `>= @onflow/fcl@0.0.76`
+
+# Overview
+
+Flow Client Library (FCL) approaches the idea of blockchain wallets on Flow in a different way than how wallets may be supported on other blockchains. For example, with FCL, a wallet is not necessarily restricted to being a browser extention or even a native application on a users device. Wallets on Flow can take the shape of being many different types applications. Since wallet applications can take on many forms, we needed to create a way for these varying applications to be able to communicate and work together.
+
+FCL acts in many ways as a protocol to facilitate communication and configuration between the different parties involved in a blockchain application. An _Application_ can use FCL to _authenticate_ users, and request _authorizations_ for transactions, as well as mutate and query the _Blockchain_. An application using FCL offers it's _Users_ a way to connect and select any number of Wallets Providers and their Wallet Services. A selected _Wallet_ configures an Applications instance of FCL with information about its services, of which the _User_ and _Application_ can interact with.
+
+In the following paragraphs we'll explore ways in which you, as a wallet developer, can integrate with FCL through providing implementataions of various FCL services. 
+
+The following services will be covered:
+- Authentication (Authn) Service
+- Authorization (Authz) Service
+- User Signature Service
+- Pre-Authz Service
+
+# Authentication
+
+In the following examples, we'll walk you through the process of building an authentication process.
+
+In FCL, wallets are configured by passing in a wallet providers authentication URL as the `discovery.wallet` config variable.
+
+As someone who is making an FCL compatible wallet, you will need to make and expose a webpage hosted at an authentication endpoint that will be rendered in an iframe.
+
+```javascript
+// IN APPLICATION
+// configuring fcl to point at a wallet looks like this
+import {config} from "@onflow/fcl"
+
+config({
+  "discovery.wallet": "your-url-that-fcl-will-render-in-iframe"
+})
```

Comment:
Can we some how mention in here that it no longer needs to be an iframe and it can be configured to follow the `HTTP/POST` or `POP/RPC` services as well as the default `IFRAME/RPC`

Reply:
I guess technically this can now all be done as a generic service... so moving the config stuff into the View methods docs might make sense... I dunno, this section as it is might not make as much sense now.